### PR TITLE
feat: Remove truncation when `stream_gen_ai_spans` is enabled

### PR DESCRIPTION
**Pull request by alexander-alderman-webb**
> **Note**: Depends on https://github.com/getsentry/sentry-python/pull/6079

### Description

Remove message popping and binary blob redaction when the `stream_gen_ai_spans` option is enabled.

Stop parametrizing truncation tests that fail when the `stream_gen_ai_spans` option is enabled.

#### Reminders
- Please add tests to validate your changes, and lint your code using `tox -e linters`.
- Add GH Issue ID _&_ Linear ID (if applicable)
- PR title should use [conventional commit](https://develop.sentry.dev/engineering-practices/commit-messages/#type) style (`feat:`, `fix:`, `ref:`, `meta:`)
- For external contributors: [CONTRIBUTING.md](https://github.com/getsentry/sentry-python/blob/master/CONTRIBUTING.md), [Sentry SDK development docs](https://develop.sentry.dev/sdk/), [Discord community](https://discord.gg/Ww9hbqr)

### CODE REVIEW

File: sentry_sdk/ai/utils.py [hunk 1]
```diff
@@ -741,6 +741,10 @@ def truncate_and_annotate_messages(
     scope: "Any",
     max_single_message_chars: int = MAX_SINGLE_MESSAGE_CONTENT_CHARS,
 ) -> "Optional[List[Dict[str, Any]]]":
+    client = sentry_sdk.get_client()
+    if client.options["_experiments"].get("stream_gen_ai_spans", False):
+        return messages
```
[the remark this answers is not in the document]
Comment (alexander-alderman-webb) [reply sentry_sdk/ai/utils.py]: that's good 👍

[the remark this answers is not in the document]
Comment (alexander-alderman-webb) [reply sentry_sdk/ai/utils.py]: that's good 👍

**Review by sentrivana — COMMENTED** [review]
Nice! Two things:
1. Does message truncation exclusively affect `gen_ai` spans currently? In other words, is it guaranteed that all spans that will become very large because of this change will be sent as v2 spans, or do we need an extra check to only lift the truncation for `gen_ai` spans specifically?
2. If we remove the parametrization from the truncation tests, will we have test coverage for the "`stream_gen_ai_spans=True` should not truncate messages" code path?

**Comment by alexander-alderman-webb** [comment]
1. That's a possible edge case, and since it's unexpected for `truncate_and_annotate_messages` to not truncate in some cases, I moved the logic to the call sites in the integrations: https://github.com/getsentry/sentry-python/pull/6260/changes/4ec3ff7e96f6ea4c2c2764b6a0d91eed4b497d08
2. By removing the parametrization we _do not_ assert that there is no truncation when `stream_gen_ai_spans=True`. I don't have strong feelings here, and removed the parametrization because I try to avoid having to many tests that assert the absence of something 🤷 . Happy to parametrize the tests again as well.

**Comment by sentrivana** [comment]
>     2. By removing the parametrization we _do not_ assert that there is no truncation when `stream_gen_ai_spans=True`. I don't have strong feelings here, and removed the parametrization because I try to avoid having to many tests that assert the absence of something 🤷 . Happy to parametrize the tests again as well.

I'm fine with removing the parametrization, but we should have _some_ test coverage for the messages being there as expected, without truncation, in the `stream_gen_ai_spans=True` case. If this is already tested by another existing test not touched by this PR, fine to leave as is. If not, would be good to add a new test for that specifically.

**Comment by alexander-alderman-webb** [comment]
Yes, that makes sense. I'll update other tests to accept two inputs messages and assert that both are present when the option is enabled (demonstrating that input truncation is disabled in this case).

File: tests/integrations/openai_agents/test_openai_agents.py [hunk 2]
```diff
@@ -327,6 +331,10 @@ async def test_agent_invocation_span_no_pii(
                     "role": "system",
                     "content": "You are a helpful assistant.",
                 },
+                {
+                    "role": "user",
+                    "content": "Message demonstrating the absence of truncation.",
+                },
```
[the remark this answers is not in the document]
Comment (alexander-alderman-webb) [reply tests/integrations/openai_agents/test_openai_agents.py:338]: sure I'll add

**Review by sentrivana — APPROVED** [review]
:shipit:
